docs/verify/_toc.json

```diff
@@ -6,9 +6,13 @@
       "url": "/verify"
     },
     {
-      "title": "Simulators overview",
+      "title": "IBM Quantum simulators",
       "url": "/verify/simulators"
```

Comment:
While we're at it, should this URL change to something like `ibm-simulators`? This URL is highly generic right now.

Since we only changed this week the URL to `verify/`, it's a good time to fix the URL.

Reply:
This is already in our URL audit PR with the suggestion to change it to cloud-based-simulators. I can change it here instead